src/modules/datepicker/directives/datepicker.directive.ts

```diff
@@ -148,6 +151,18 @@ export class SuiDatepickerDirective
         }
     }
 
+    public ngOnInit():void {
```

Comment:
Rather than going this way, make use of the pre existing `currentDate` property of the `CalendarService`.

In the datepicker directive, add the following just before you set the service selected date:
```typescript
this.componentInstance.service.currentDate = this.initialDate || new Date()
```

Then in the calendar service, change `reset` to something like:
```typescript
public reset():void {
    this.currentView = this.config.mappings.finalView;

    if (!this._selectedDate) {
        let current = this.currentDate.getTime();
        if (this._minDate) {
            current = Math.max(current, this._minDate.getTime());
        }
        if (this._maxDate) {
            current = Math.min(current, this._maxDate.getTime());
        }

        this.currentDate = new Date(current);
        this.config.updateBounds(this.currentDate);

        this.currentView = this.config.mappings.initialView;
    }
}
```

That should do the trick without needing to do any `null` checks etc 😄